### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by whiterook6**
So ... Big changes here:
1. Score is now an incrementing value in the Player class.
2. Game is now a thing that keeps track of players.
   2.1. Game ticks every 5 seconds, and ticks all players, which update their own scores and send them down to clients, along with a score speed (how many spins per minute?) This allows for effects like slowing scores, etc.
3. Server initials and team are now in the player class, not the socket.
4. Chatting is working again: global chat, system chat, and team chat.
   Old libs and server left in for now; can be removed eventually.

### CODE REVIEW

File: lib-old/chat.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		socket.on('message.global', function(data){
+			if (!data || !data.text || !socket.initials){ // can't send nothing or send without initials.
+				return;
+			}
+
+			if ((Timer.now() - socket.last_message) < 10){
+				return;
+			}
+
+			var message_re = /#([a-zA-Z0-9]{3}) (.+)/, // match #BOB you guys suck -> match[0]='BOB', match[1]='you guys suck'
```
Comment (whiterook6) [inline lib-old/chat.js]: match 1, match 2

File: lib-old/chat.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+				team,
+				message,
+				packet = {
+					from: {
+						initials: socket.initials //,
+						// team: socket.team
+					}
+				};
+
+			if (socket.team){
```
Comment (whiterook6) [inline lib-old/chat.js]: socket.player.team

File: lib/chat.js [hunk 3]
```diff
@@ -1,55 +1,124 @@
...
+			packet.from.team = player.team;
+		}
+
+		player.socket.broadcast.to(to_team).emit('message.team', packet);
+	},
+
+	send_from_system: function(text){
+		Logger.debug('Chat.send_from_system');
+		if (!text){
+			Logger.warning('Sending message without text');
```
Comment (whiterook6) [inline lib/chat.js:68]: make sure to return from this line.

File: lib/chat.js [hunk 4]
```diff
@@ -1,55 +1,124 @@
...
+			Logger.debug('Event: message.global');
+
+			if (!data || !data.text || !socket.player || !socket.player.initials){ // can't send nothing or send without initials.
 				return;
 			}
 
-			if ((Timer.now() - socket.last_message) < 10){
+			var player = socket.player;
+
+			if (!player.last_message){
```
Comment (whiterook6) [inline lib/chat.js:97]: make sure to do flood detection for team messages as well. Also return an event if the user is flooded, so the UI can show this.

**Comment by mayppong** [comment]
I would rather you just remove the old stuff. There's no point in keeping them since we're using a version control already.

Would you be against just coding in ES6? If you're going to declare things like constructor, you might enjoy ES6 a lot more.

The code seems a lot cleaner in general so I'm happy about that. Again, I'm not super excited about all the return statement and how often the Logger is being called though. I feel like utilizing exceptions would work much better.

In any case, I'm will to accept this as is if you prefer, so we can continue develop on it regardless of the decisions over the above concerns.

File: lib/chat.js (hunk 4 above)
Comment (whiterook6) [inline lib/chat.js:97]: done

File: lib/chat.js (hunk 3 above)
Comment (whiterook6) [inline lib/chat.js:68]: done

**Pull request merged**